packages/ui/src/components/Charts/helpers/getPayloadConfigFromPayload.ts

```diff
@@ -0,0 +1,26 @@
+import type { ChartConfig } from "../types";
+
+export function getPayloadConfigFromPayload(config: ChartConfig, payload: unknown, key: string) {
```

Comment:
We can probably make this simpler but using a type guard for records like this:

```ts
function isRecord(value: unknown): value is Record<string, unknown> {
  return typeof value === 'object' && value !== null && !Array.isArray(value);
}
```


and then simplify the function without as many type casts

```ts
export function getPayloadConfigFromPayload(config: ChartConfig, payload: unknown, key: string) {
  if (!isRecord(payload)) {
    return undefined;
  }

  const payloadPayload = isRecord(payload.payload) ? payload.payload : undefined;

  const configLabelKey = 
    typeof payload[key] === "string" ? payload[key] :
    payloadPayload && typeof payloadPayload[key] === "string" ? payloadPayload[key] :
    key;

  return config[configLabelKey as keyof ChartConfig] ?? config[key as keyof ChartConfig];
}
```

it should work (™️)

Reply:
[5ac177f](https://github.com/synopsisapp/symbiosis-ui/pull/9/commits/5ac177f000c070774e9b8f28d4c167269f22eb2d)